src/main/java/com/khomishchak/cryptoportfolio/repositories/SelfGoalRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.khomishchak.cryptoportfolio.repositories;
+
+import com.khomishchak.cryptoportfolio.model.goals.SelfGoal;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
+
+public interface SelfGoalRepository extends JpaRepository<SelfGoal, Long> {
+
+    List<SelfGoal> findAllByUserId(Long userId);
+
+    @Query(value = "SELECT * FROM self_goals sf WHERE NOW() > sf.end_date AND sf.is_closed = false", nativeQuery = true)
+    List<SelfGoal> getAllOverdueGoals();
```

Comment:
Why do you use native query here? If there is no way to write it in the JPQL does not better to handle it on the BE side?